### PR TITLE
feat(react-drawer): implement "prevent close on click outside" feature

### PR DESCRIPTION
**Pull request by marcosmoura**
See title

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/57eef05cefcaae10aa48a074b2b0851d4dc43763'>57eef05cefcaae10aa48a074b2b0851d4dc43763</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-drawer/src/components/Drawer/useDrawer.ts [hunk 1]
```diff
@@ -49,6 +61,7 @@ export const useDrawer_unstable = (props: DrawerProps, ref: React.Ref<HTMLElemen
     position = 'left',
     size = 'small',
     modal = true,
+    lightDismiss = true,
```
Comment (ValentinaKozlova) [inline packages/react-components/react-drawer/src/components/Drawer/useDrawer.ts:64]: Have you checked that the default behavior works? Looks like the Drawer is not closing by default anymore

**Comment by Hotell** [comment]
it's good practice to include UI demo/video for feature additions within PR description if possible. 🙏 can we add that ?

File: packages/react-components/react-drawer/src/components/Drawer/useDrawer.ts (hunk 1 above)
Comment (marcosmoura) [reply packages/react-components/react-drawer/src/components/Drawer/useDrawer.ts:64]: This is now fixed. I added also tests to ensure this.

**Review by ValentinaKozlova — APPROVED** [review]

**Pull request merged**